code_conventions.md

```diff
@@ -1,108 +1,75 @@
-Example:
-~~~~~~~~~~~~~~~{.json}
-// Temperature compensation used in marlin for control and reporting according to the following formula...
-// compensated = (measured * factor) + offset
-"temperature_compensation_hotends": {
-    // No compensation
-    "0": {
-        "factor": 1.0,
-        "offset": 0.0
-    },
-    // At measured 210.0 degrees C there is a discrepancy of -2.4 degrees C
-    // Begin compensation starting at 50.0 degrees
-    // Gain is 2.4 / (210.0 - 50.0)
-    "1": {
-        "factor": 0.985, // = 1.0 - gain
-        "offset": 0.75   // = 50.0 * gain
-    }
-},
-~~~~~~~~~~~~~~~
-
-Logging
-----
-Logging should be done on a few levels: 
-* DEBUG: Verbose logging -> logging data that is useful to debug parts of the code being run
-* INFO: Logging -> logging information that can be seen as feedback to a user on his/her actions (acties gebruiker en acties systeem)
-* WARN: Warning message are an indication to the user that something is not entirely right, but might not yet be a big issue
-* ERROR: When a function cannot continue at all, but it doesnt mean that the application has to stop working.
-* CRITICAL An error is a situation that cannot be overcome without external influence. This implies that the application is very likely to stop working at all
-
-With the above guidelines, try to use common sense in what log level to use. Especially with DEBUG and INFO the border is not very clear.
-
-*** New ***
-uranium logging
-python logging
-[easylogging++ (boost)]
-compile time log level?
+# Code Conventions
+This document describes the code conventions and guidelines to be followed in all Ultimaker code, regardless the programming environment / language.
+
+In certain cases specific rules might apply, depending on the programming language used, but in general all the conventions apply to all programming and/or scripting languages.
+
+Do note that not all the code convention described here have been fully implemented yet. However, any newly committed code MUST follow the conventions below, unless the document of the specific language overrides one or more of these rules.
+
+When contributing to other (external / open source) projects, the coding guidelines of the respective projects MUST be followed.
+
+## Commenting
+There are 4 kinds of comments that can be used;
+* Comments for interface / signature documentation purposes (see Doxygen Commenting in different document)
+* Comments to indicate that something needs to be examined at a later stage and possibly be refactored. When this is done, it MUST be accompanied by a Jira issue / story
+* Comments that explain a workaround. In these cases, the documentation must state why a work around is required instead of a "real" solution.
+* Comments to clarify implementation decisions (eg; Why is something done in a certain way)
+
+Comments in general never should state the obvious. If the comment is simply rephrasing the code, it should be left out.
```

Comment:
This is contradictory with the doxygen rule as adding doxygen for everything re-states typing and variable name information.
Please indicate which is the overriding factor.

Reply:
The doxygen comments part needs an overhaul, but that's outside the scope of this cleanup.